test/system_metrics_e2e_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+End to end tests for the system_metrics_collector ROS2 package.
+
+These tests launch the talker_listener_example and use ROS2 to inspect if the
+expected nodes are active and publishing data. To run these tests ensure that
+ROS2 is installed, with the system_metrics_collector package the relevant
+setup.bash has been sourced.
+"""
+
+import logging
+import signal
+import subprocess
+import sys
+
+import rclpy
+
+import test_helpers
+
+
+# Expected outputs
+EXPECTED_LIFECYCLE_NODES = frozenset(['/linux_system_cpu_collector',
+                                      '/linux_system_memory_collector',
+                                      '/listener_process_cpu_node',
+                                      '/listener_process_memory_node',
+                                      '/talker_process_cpu_node',
+                                      '/talker_process_memory_node'])
+EXPECTED_REGULAR_NODES = frozenset(['/listener', '/talker'])
+EXPECTED_LIFECYCLE_STATE = 'active [3]'
+EXPECTED_TOPIC = '/system_metrics'
+# Commands to execute
+LAUNCH_COMMAND = 'ros2 launch system_metrics_collector talker_listener_example.launch.py'
+# Test constants
+TIMEOUT_SECONDS = 30
+RETURN_VALUE_FAILURE = 1
+RETURN_VALUE_SUCCESS = 0
+EXPECTED_NUMBER_OF_MESSAGES_TO_RECEIVE = 5
+
+
+def setup_logger() -> None:
```

Comment:
This can be part of the helper functions. Here's an example:

logger.py
```

import logging
import sys

def setup_logger(name):
    """Format and setup the logger."""
    logger = logging.getLogger(name)
    # setting to debug for end to end test soak
    logger.setLevel(logging.DEBUG)

    handler = logging.StreamHandler(sys.stdout)
    handler.setLevel(logging.DEBUG)
    formatter = logging.Formatter('[%(module)s] [%(levelname)s] [%(asctime)s]: %(message)s')
    handler.setFormatter(formatter)
    logger.addHandler(handler)

    return logger
```

log_hi.py
```

import logging
import logger

def hello():
	my_logger = logger.setup_logger('my_logger')
	my_logger.debug('hi debug')
	my_logger.info('hi info')

	get_logger = logging.getLogger('my_logger')
	get_logger.debug('hi debug')
	get_logger.info('hi info')
	
if __name__ == "__main__":
	hello()
```

Output:
```
dbbonnie@f01898a4625e:~/tmpPy$ python3 log_hi.py 
[log_hi] [DEBUG] [2020-04-13 16:13:37,984]: hi debug
[log_hi] [INFO] [2020-04-13 16:13:37,984]: hi info
[log_hi] [DEBUG] [2020-04-13 16:13:37,985]: hi debug
[log_hi] [INFO] [2020-04-13 16:13:37,985]: hi info
```

Note how you can either return the logger from the method or simply get the logger by name.

Reply:
Moved logging setup to helper module.